raiden/tests/integration/network/transport/rif_comms/test_client.py

```diff
@@ -23,92 +24,173 @@ def comms_nodes(amount_of_nodes) -> {int, CommsNode}:
 
 
 @pytest.mark.parametrize("amount_of_nodes", [1])
-def test_locate_own_peer_id(comms_nodes):
-    comms_node = comms_nodes[1]
+def test_subscribe_to_invalid(comms_nodes):
+    client = comms_nodes[1].client
+    # subscribe to unregistered address
+    topic_id, _ = client.subscribe_to(generate_address())
 
-    # attempt to locate self
-    assert comms_node.client._get_peer_id(comms_node.address) is not ""
+    # FIXME: this should probably throw an exception rather than an empty result
+    assert topic_id is ""
 
 
-# FIXME: causes ERR_NO_PEERS_IN_ROUTING_TABLE in comms node although it passes
 @pytest.mark.parametrize("amount_of_nodes", [1])
-def test_locate_unregistered_peer_id(comms_nodes):
-    comms_node = comms_nodes[1]
+def test_subscribe_to_self(comms_nodes):
+    client, address = comms_nodes[1].client, comms_nodes[1].address
+
+    topic_id, _ = client.subscribe_to(address)
```

Comment:
Why are we ignoring a value here, since this is a test we should check all values right?

Reply:
> Why are we ignoring a value here, since this is a test we should check all values right?

the value ignored here is the stream for subscriptions, which is only used for receiving or sending messages through it.

since this test only checks that a subscription is successfully made, it only cares that there is a non-empty topic id for the subscription.

moreover, since no messages are expected to be received or sent through the aforementioned stream, there would be no point in checking it; we'd just wait for a message which would never come.

there are tests which _do_ check this variable, namely: `test_send_message_subscription`, `test_send_message_self`, and `test_send_message_peers`. in contrast to this test, these do not care about the topic id, precisely for the reason of having each test test a different thing.

Reply:
Sounds like subscribe_to does more than one thing, get the stream and the topic id.

Reply:
> Sounds like subscribe_to does more than one thing, get the stream and the topic id.

correct. this is the way the function is programmed in the pubsub node